### ISSUE TITLE
No dtype for array of arrays into tf.keras.Input

### ISSUE THREAD

**Issue by Chasearmer** [tensorflow/tensorflow#19864]
I would like to input an array of arrays into `tf.keras.Input()`, however that requires me to specify a dtype to ensure it doesn't default to expecting a float. However it doesn't appear as though there is a viable dtype for an array of arrays. For an array by itself, `dtype=float32_ref` appears to work, however this does not work for an array of arrays. `tf.keras.backend.dtype()` throws an error if I attempt to put in an array to see it's type as well. Does this dtype not exist?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Chasearmer** [comment]
Ubuntu 16.04
Tensorflow 1.5.0

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by cy89** [comment]
@fchollet can you comment on array-of-arrays support?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 32 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 48 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 93 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 108 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 123 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 138 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jvishnuvardhan** [comment]
@Chasearmer Is this still an issue with the latest TF versions? If yes, can you provide a standalone code to reproduce the issue? Thanks!

**Comment by jvishnuvardhan** [comment]
The following issue is somewhat related which may help.
https://github.com/keras-team/keras/issues/15778
